### PR TITLE
EZP-31735: As a Developer I'd like direct upgrade path from 1.13LTS to 2.5LTS

### PR DESCRIPTION
**Pull request by andrerom**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31735](https://jira.ez.no/browse/EZP-31735) `As a Developer I'd like direct upgrade paths from LTS to LTS`
| **Bug/Improvement**| yes
| **Target version** | `7.5`
| **BC breaks**      | no
| **Tests pass**     | N/A
| **Doc needed**     | yes _(to reach goal of story)_

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Review by alongosz — COMMENTED** [review]
Hmm, if I understood your intention right, I would expect rather `to-7.5.x.sql` or `to-7.5.latest.sql`. It would be clear that this contains all the needed changes. Or is developer still expected to run all `*-to-7.5.*.sql` scripts?

File: data/update/mysql/dbupdate-6.13.0-to-7.5.7.sql [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `owner_id` int(11) NOT NULL DEFAULT 0,
+  `is_pending` tinyint(1) NOT NULL DEFAULT '1',
+  `type` varchar(128) NOT NULL DEFAULT '',
+  `created` int(11) NOT NULL DEFAULT 0,
+  `data` blob,
+  PRIMARY KEY (`id`),
+  KEY `eznotification_owner` (`owner_id`),
+  KEY `eznotification_owner_is_pending` (`owner_id`, `is_pending`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (alongosz) [inline data/update/mysql/dbupdate-6.13.0-to-7.5.7.sql]: utf8mb4?
```suggestion
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

**Review by alongosz — CHANGES_REQUESTED** [review]
One remark regarding the MySQL script itself:

**Comment by andrerom** [comment]
> Hmm, if I understood your intention right, I would expect rather to-7.5.x.sql or to-7.5.latest.sql

No that is not what I intended imply here, files like this can not be moving target, that will just guarantee doc and links they point to will get out of sync

So it should have been `data/update/mysql/dbupdate-6.13.0-to-7.5.0.sql`, however as we never added it once we made 2.5LTS I was thinking we could provide a direct one to .7

On the other side, maybe I should change it to be `data/update/mysql/dbupdate-6.13.0-to-7.5.0.sql`, to make sure people always expect to have to followup update doc as natural followup _(as there will be future updates, and we won't bother to continuously update the migration doc I'm using this in)_

WDYT?

**Comment by andrerom** [comment]
Updated to go from 6.13.0 to 7.5.0 as the others. /cc @vidarl

File: data/update/mysql/dbupdate-5.4.0-to-6.13.0.sql [hunk 2]
```diff
@@ -2,6 +2,9 @@ SET default_storage_engine=InnoDB;
 -- Set storage engine schema version number
 UPDATE ezsite_data SET value='6.13.0' WHERE name='ezpublish-version';
 
+-- IMPORTANT:
+-- * Only covers database schema changes in eZ Platform, for other bundles you use (enterprise or otherwise), see doc!
```
Comment (glye) [inline data/update/mysql/dbupdate-5.4.0-to-6.13.0.sql]: Doc link maybe? Or if there's no general doc for this and you're referring to doc for each bundle, then ok as is, or "see doc for each bundle".

File: data/update/postgres/dbupdate-6.13.0-to-7.5.0.sql [hunk 3]
```diff
@@ -0,0 +1,298 @@
+UPDATE ezsite_data SET value='7.5.0' WHERE name='ezpublish-version';
+
+-- IMPORTANT:
+-- * Only covers database schema changes in eZ Platform, for other bundles you use (enterprise or otherwise), see doc!
+
+
+-- 7.1.0/6.13.0 - 7.2.0
```
Comment (glye) [inline data/update/postgres/dbupdate-6.13.0-to-7.5.0.sql]:
```suggestion
-- 6.13.0/7.1.0 - 7.2.0
```
maybe?

**Review by glye — APPROVED** [review]

File: data/update/mysql/dbupdate-6.13.0-to-7.5.0.sql [hunk 4]
```diff
@@ -0,0 +1,180 @@
+UPDATE ezsite_data SET value='7.5.0' WHERE name='ezpublish-version';
+
+-- IMPORTANT:
+-- * If you use DFS (cluster), also make sure to apply "dbupdate-7.1.0-to-7.2.0-dfs.sql"
+-- * Only covers database schema changes in eZ Platform, for other bundles you use (enterprise or otherwise), see doc!
+
+
+-- 7.1.0/6.13.0 - 7.2.0
```
Comment (andrerom) [inline data/update/mysql/dbupdate-6.13.0-to-7.5.0.sql]:
```suggestion
-- 6.13.0/7.1.0 - 7.2.0
```

File: data/update/mysql/dbupdate-5.4.0-to-6.13.0.sql (hunk 2 above)
Comment (andrerom) [reply data/update/mysql/dbupdate-5.4.0-to-6.13.0.sql]: Better? _(if so I'll apply all over)_
```suggestion
-- * This only covers eZ Platform kernel, see doc for all steps (incl. database schema) needed when updating: https://doc.ezplatform.com/en/2.5/updating/updating_ez_platform/
```

Comment (glye) [reply data/update/mysql/dbupdate-5.4.0-to-6.13.0.sql]: Looks good.

File: data/update/mysql/dbupdate-6.13.0-to-7.5.0.sql [hunk 5]
```diff
@@ -0,0 +1,180 @@
+UPDATE ezsite_data SET value='7.5.0' WHERE name='ezpublish-version';
+
+-- IMPORTANT:
+-- * If you use DFS (cluster), also make sure to apply "dbupdate-7.1.0-to-7.2.0-dfs.sql"
+-- * Only covers database schema changes in eZ Platform, for other bundles you use (enterprise or otherwise), see doc!
```
Comment (andrerom) [inline data/update/mysql/dbupdate-6.13.0-to-7.5.0.sql]:
```suggestion
-- * This only covers eZ Platform kernel, see doc for all steps (incl. database schema) needed when updating: https://doc.ezplatform.com/en/2.5/updating/updating_ez_platform/
```

File: data/update/postgres/dbupdate-5.4.0-to-6.13.0.sql [hunk 6]
```diff
@@ -1,6 +1,9 @@
 -- Set storage engine schema version number
 UPDATE ezsite_data SET value='6.13.0' WHERE name='ezpublish-version';
 
+-- IMPORTANT:
+-- * Only covers database schema changes in eZ Platform, for other bundles you use (enterprise or otherwise), see doc!
```
Comment (andrerom) [inline data/update/postgres/dbupdate-5.4.0-to-6.13.0.sql]:
```suggestion
-- * This only covers eZ Platform kernel, see doc for all steps (incl. database schema) needed when updating: https://doc.ezplatform.com/en/2.5/updating/updating_ez_platform/
```

File: data/update/postgres/dbupdate-6.13.0-to-7.5.0.sql [hunk 7]
```diff
@@ -0,0 +1,298 @@
+UPDATE ezsite_data SET value='7.5.0' WHERE name='ezpublish-version';
+
+-- IMPORTANT:
+-- * Only covers database schema changes in eZ Platform, for other bundles you use (enterprise or otherwise), see doc!
```
Comment (andrerom) [inline data/update/postgres/dbupdate-6.13.0-to-7.5.0.sql]:
```suggestion
-- * This only covers eZ Platform kernel, see doc for all steps (incl. database schema) needed when updating: https://doc.ezplatform.com/en/2.5/updating/updating_ez_platform/
```

**Comment by andrerom** [comment]
@alongosz Better now? _(It's now `dbupdate-6.13.0-to-7.5.0.sql` to eliminate any indications it will be kept updated for minor versions)_

**Review by vidarl — APPROVED** [review]

**Comment by vidarl** [comment]
One thing though. `dbupdate-6.13.0-to-7.5.0.sql` doesn't mention anything in regards to [dbupdate-7.1.0-to-7.2.0-dfs.sql](https://github.com/ezsystems/ezpublish-kernel/blob/7.5/data/update/mysql/dbupdate-7.1.0-to-7.2.0-dfs.sql). It should be mentioned in any doc page using `dbupdate-6.13.0-to-7.5.0.sql` that you'll also need to run `dbupdate-7.1.0-to-7.2.0-dfs.sql` if you're using cluster

**Comment by andrerom** [comment]
@vidarl I kind of tried to cover this inline her in the file, so feel free to propose improvements if you see any:
```
-- IMPORTANT:
-- * If you use DFS (cluster), also make sure to apply "dbupdate-7.1.0-to-7.2.0-dfs.sql"
-- * This only covers eZ Platform kernel, see doc for all steps (incl. database schema) needed when updating: https://doc.ezplatform.com/en/2.5/updating/updating_ez_platform/
```

**Comment by andrerom** [comment]
@vidarl indeed it needs to be covered in doc as well so made note of it here: https://github.com/ezsystems/developer-documentation/pull/612


For the inline info in PR here it's ok or?
```
-- IMPORTANT:
-- * If you use DFS (cluster), also make sure to apply "dbupdate-7.1.0-to-7.2.0-dfs.sql"
-- * This only covers eZ Platform kernel, see doc for all steps (incl. database schema) needed when updating: https://doc.ezplatform.com/en/2.5/updating/updating_ez_platform/
```

**Comment by vidarl** [comment]
> For the inline info in PR here it's ok or?

Yeah, it's ok. ( I missed it when reviewing yesterday, but I see it's there now )

**Pull request merged**